### PR TITLE
API/DOC: Some API cleanup previously discussed and new API docs

### LINKED ISSUE #207: Rethink Header.events and Header.stream
Believe or not, @tacaswell is advocating for a name change. I'm just the note-taker here. Adjust my reputation among my peers accordingly.

* Header.events -> Header.stream
* Header.stream -> Header.documents

Also, the new Header.stream should *require* its stream_name arg and disallow `ALL`. That is, it should return rows from a table. That analogy implies that the rows have homogeneous columns.

### PR DESCRIPTION
**Pull request by danielballan**
The databroker documentation still needs a lot of love, but this
is a good start.

Also, closes #207

### CODE REVIEW

File: databroker/broker.py [hunk 1]
```diff
@@ -351,12 +405,42 @@ def clear_filters(self, **kwargs):
...
 
     def __getitem__(self, key):
-        """Do-What-I-Mean slicing"""
+        """
+        Search runs based on recently, unique id, or counting number scan_id.
+
+        This function returns a :class:`Header` object (or a list of them, if
+        the input is a list or slice). Each Header encapsulates the metadata
+        for a run -- start time, instruments used, and so on, and provides
+        methods for loading the data. In
```
Comment (tacaswell) [inline databroker/broker.py]: Missing sentence.

File: doc/source/api.rst [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+===============
+
+You can instantiate a :class:`Broker` by passing it a dictionary of
+configuration or by providing the name of a configuration file on disk.
+
+.. autosummary::
+   :toctree: _as_gen
+   :nosignatures:
+
+   Broker.from_config
```
Comment (licode) [inline doc/source/api.rst:31]: I think this is a much convenient interface for configuration setup at different beamlines.

File: databroker/broker.py [hunk 3]
```diff
@@ -418,53 +502,58 @@ def dynamic_alias(self, key, func):
...
+        queries using the MongoDB query syntax.
 
         Parameters
         ----------
         text_search : str, optional
             search full text of RunStart documents
-        start_time : time-like, optional
-            Include Headers for runs started after this time. Valid
-            "time-like" representations are:
-                - float timestamps (seconds since 1970), such as time.time()
```
Comment (tacaswell) [inline databroker/broker.py:435]: is'time-like' defined someplace else?

**Comment by licode** [comment]
waiting for the doc. good work 👍

File: databroker/broker.py (hunk 3 above)
Comment (licode) [reply databroker/broker.py:435]: something like this:  
https://github.com/NSLS-II/databroker/blob/master/databroker/headersource/core.py#L767

File: databroker/broker.py [hunk 4]
```diff
@@ -510,7 +599,7 @@ def fill_event(self, event, inplace=True,
     def get_events(self, headers, fields=None, stream_name=ALL, fill=False,
```
Comment (tacaswell) [inline databroker/broker.py:650]: This function now very problematic.   It should probably take exactly 1 header and switch to a single stream.

**Pull request merged**